File: src/PomodoroWindowsTimer.Installer/Configuration/DependencyInjectionExtensions.cs
using Microsoft.Extensions.DependencyInjection;
using PomodoroWindowsTimer.Installer.Abstractions;

namespace PomodoroWindowsTimer.Installer.Configuration;

public static class DependencyInjectionExtensions
{
    public static void AddInstallerServices(this IServiceCollection services)
    {
        services
            .AddOptions<PwtGitHubClientOptions>()
            .BindConfiguration(PwtGitHubClientOptions.SectionName)
            .ValidateDataAnnotations()
            .ValidateOnStart();

        services.AddTransient<IPwtGitHubClient, PwtGitHubClient>();
    }
}

File: src/PomodoroWindowsTimer.ElmishApp/DependencyInjectionExtensions.fs
namespace PomodoroWindowsTimer.ElmishApp

open System.Runtime.CompilerServices
open Microsoft.Extensions.DependencyInjection
open Microsoft.Extensions.DependencyInjection.Extensions
open PomodoroWindowsTimer.Abstractions

[<Extension>]
type DependencyInjectionExtensions () =
    [<Extension>]
    static member AddElmishAppServices(services: IServiceCollection) =
        services.TryAddSingleton<WorkEventStore>(fun sp ->
            let repositoryFactory = sp.GetRequiredService<IRepositoryFactory>()
            WorkEventStore.init repositoryFactory
        )
